packages/dom-adapters/src/BlockToolAdapter/index.ts

```diff
@@ -133,40 +195,82 @@ export class BlockToolAdapter {
       case InputType.DeleteSoftLineForward:
       case InputType.DeleteWordBackward:
       case InputType.DeleteWordForward: {
-        this.#model.removeText(this.#blockIndex, key, start, end);
-
+        if (nativeInput) {
+          this.#handleDeleteInNativeInput(event, input as HTMLInputElement | HTMLTextAreaElement, key);
+        } else {
+          this.#handleDeleteInContentEditable(event, input, key);
+        }
         break;
       }
 
       default:
     }
-  }
+  };
 
-  /**
-   * Handles model update events and updates DOM
-   *
-   * @param event - model update event
-   * @param input - attched input element
-   * @param key - data key input is attached to
-   * @param caretAdapter - caret adapter instance
-   */
-  #handleModelUpdate(event: ModelEvents, input: HTMLElement, key: DataKey, caretAdapter: CaretAdapter): void {
+  #handleModelUpdateNative = (event: ModelEvents, input: HTMLInputElement | HTMLTextAreaElement, key: DataKey): void => {
+    if (!(event instanceof TextAddedEvent) && !(event instanceof TextRemovedEvent)) {
+      return;
+    }
+
+    const [, dataIndex, blockIndex] = event.detail.index;
```

Comment:
We use `dataKey` in other places, let's keep it for consistency

```suggestion
    const [, dataKey, blockIndex] = event.detail.index;
```